charachorder/device.py

```diff
@@ -28,15 +28,15 @@
 
 pid_mapping = {}
 vid_mapping = {
-    0x239A: ("Adafruit", "M0"),
-    0x303A: ("Espressif", "S2"),
+    0x239A: "Adafruit",
+    0x303A: "Espressif",
 }
 
 
-def allowed_product_ids(*product_ids: int):
+def allowed_product_ids(*product_ids: (int, str)):
```

Comment:
**suggestion:** Shall we consider `product_ids: {int: str}`? A dict seems easier to type and use than key-value pairs.

Reply:
Dict dereferencing for decorators was messy, feel free to change it as long as you can make it work

Reply:
Were you referring to dereferencing a `**kwargs`-like dict? I do notice that it currently also uses a `*args`-like sequence. I would personally prefer to use only one parameter which is a container. Such as:
```suggestion
def allowed_product_ids(product_ids: {int, str}):
```
and then it would be used like this:

```python
@allowed_product_ids({
    0x8189: "S2",
    0x818A: "S2 - UF2 Bootloader",
})
class CharaChorderEngine(CharaChorder):
```

> ... as long as you can make it work

Admittedly, I haven't tested it in this code base (don't know how), but the usage above should work.

Reply:
We shouldn't be messing with any of this IMO. Will get to it once I have the time for a review